core/tests/Test.Microsoft.Identity.LabInfrastructure/KeyVaultSecretsProvider.cs

```diff
@@ -119,5 +147,27 @@ private async Task<string> AuthenticationCallbackAsync(string authority, string
 
             return authResult?.AccessToken;
         }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposed)
+                return;
+
+            if (disposing)
+            {
+                //No managed resources to dispose.
+            }
+
+            // Disposing unmanaged resources.
+            _keyVaultClient.Dispose();
```

Comment:
According to the documentation for File.ReadAllText()
"The file handle is guaranteed to be closed by this method, even if exceptions are raised."
With that, I believe we can safely assume that the resource is disposed properly.

https://docs.microsoft.com/en-us/dotnet/api/system.io.file.readalltext?redirectedfrom=MSDN&view=netframework-4.7.2#System_IO_File_ReadAllText_System_String_

Reply:
Also, the data within the file will be cleared via VSTS task

Reply:
This ask was about ensuring the file is deleted from the device it's created on and where it may be sent to. Is it deleted both from the build and from the device which runs the tests?

Reply:
as said this is not about disposing the returned object from reading the string. but the text file itself (imo)

---
In reply to: [224925931](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1235#discussion_r224925931) [](ancestors = 224925931)

Reply:
Updated comments to make the disposal of this file more clear.